src/Time/TimeSteppers/AdamsBashforthN.hpp

```diff
@@ -376,22 +376,36 @@ void AdamsBashforthN::update_u(
     const gsl::not_null<Vars*> u,
     const gsl::not_null<History<Vars, DerivVars>*> history,
     const TimeDelta& time_step) const noexcept {
+  ASSERT(history->size() >= history->integration_order(),
+         "Insufficient data to take an order-" << history->integration_order()
+         << " step.  Have " << history->size() << " times, need "
+         << history->integration_order());
+  history->mark_unneeded(
```

Comment:
does moving the `mark_unneeded` also make FSAL support easier?

Reply:
Yes.  When I do this update for the substep methods I'll be able to set up FSAL, and that will let the DP5 dense output test pass.  (Commented out in this PR.)